File: src/csharp/Intel/Iced.UnitTests/Intel/DecoderTests/CodeValueReader.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

using System;
using System.Collections.Generic;
using System.IO;
using Iced.Intel;

namespace Iced.UnitTests.Intel.DecoderTests {
	static class CodeValueReader {
		public static HashSet<Code> Read(string name) {
			var filename = PathUtils.GetTestTextFilename(name, "Decoder");
			int lineNumber = 0;
			var hash = new HashSet<Code>();
			foreach (var line in File.ReadLines(filename)) {
				lineNumber++;
				if (line.Length == 0 || line[0] == '#')
					continue;
				var value = line.Trim();
				if (CodeUtils.IsIgnored(value))
					continue;
				if (!ToEnumConverter.TryCode(value, out var code))
					throw new InvalidOperationException($"Error parsing Code file '{filename}', line {lineNumber}: Invalid value: {value}");
				hash.Add(code);
			}
			return hash;
		}
	}
}

File: src/java/iced-x86/src/test/java/com/github/icedland/iced/x86/dec/CodeValueReader.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.dec;

import java.util.ArrayList;
import java.util.HashSet;

import com.github.icedland.iced.x86.CodeUtils;
import com.github.icedland.iced.x86.FileUtils;
import com.github.icedland.iced.x86.PathUtils;
import com.github.icedland.iced.x86.ToCode;

final class CodeValueReader {
	public static HashSet<Integer> read(String name) {
		String filename = PathUtils.getTestTextFilename("Decoder", name);
		ArrayList<String> lines = FileUtils.readAllLines(filename);
		HashSet<Integer> hash = new HashSet<Integer>();
		int lineNumber = 0;
		for (String line : lines) {
			lineNumber++;
			if (line.length() == 0 || line.charAt(0) == '#')
				continue;
			String value = line.trim();
			if (CodeUtils.isIgnored(value))
				continue;
			Integer code = ToCode.tryGet(value);
			if (code == null)
				throw new UnsupportedOperationException(
						String.format("Error parsing Code file '%s', line %d: Invalid value: %s", filename, lineNumber, value));
			hash.add(code);
		}
		return hash;
	}
}
